q2_coordinates/tests/test_quadtree.py

```diff
@@ -0,0 +1,159 @@
+import pandas as pd
+import pandas.testing as pdt
+import numpy as np
+import skbio
+from .. import qtrees
+from io import StringIO
+import numpy.testing as npt
+import unittest
+#test every path through code including exception raising for comprehension
+
+class BasicTest(unittest.TestCase):
+        def setUp(self):
+            #setup function for data creation
+
+            #set the starting input
+            test_points = [['test_id_sw1', -180.0, -90.0],
+                ['test_id_nw1', -180, 90],
+                ['test_id_ne1', 180, 90],
+                ['test_id_se1', 180, -90],
+                ['test_id_sw2', -91.0, -44.0],
+                ['test_id_nw2', -91, 44],
+                ['test_id_ne2', 91, 44], 
+                ['test_id_se2', 91, -44]]
+            test_df =pd.DataFrame(test_points,
+                columns = ['SampleID', 'longitude', 'latitude'])
+            
+            test_df['latitude'] = test_df['latitude'] + 90
+
+            test_df['longitude'] = test_df['longitude'] + 180
+
+            self.test_df = test_df.set_index('SampleID')
+            
+            #create answer dataframe
+            correct_point = [['test_id_sw1', '3.', '3.3.'], 
+                ['test_id_nw1', '1.', '1.1.'],
+                ['test_id_ne1', '2.', '2.2.'],
+                ['test_id_se1', '4.', '4.4.'],
+                ['test_id_sw2', '3.', '3.1.'],
+                ['test_id_nw2', '1.', '1.3.'],
+                ['test_id_ne2', '2.', '2.4.'],
+                ['test_id_se2', '4.', '4.2.']]
+            
+            #Set the correct dataframe
+            correct_dataframe = pd.DataFrame(correct_point, 
+                columns=['SampleID', 'H1', 'H2'])
+            self.correct_dataframe = correct_dataframe.set_index('SampleID')
+            
+            #set the correct tree
+            #this needs to be fixed
+            self.correct_tree = skbio.TreeNode.read(StringIO(
+                "((('test_id_sw1')'3.3',('test_id_sw2')'3.1')'3.', (('test_id_nw1')'1.1.',('test_id_nw2')'1.3.')'1.', (('test_id_se1')'4.4.', ('test_id_se2')'4.2.')'4.', (('test_id_ne1')'2.2.',('test_id_ne2')'2.4.')'2.')root;"))
```

Comment:
might be easier to express as:

```python
tree_str = ("foo bar baz"
            "bing asasdas")
```

...or alternatively, it would be appropriate to drop a `# noqa` on the end